yaml-tests/src/main/java/com/apple/foundationdb/relational/yamltests/server/ExternalServer.java

```diff
@@ -84,13 +86,12 @@ public String getVersion() {
 
     public void start() throws Exception {
         File jar;
-        if (jarName == null) {
-            final File externalDirectory = new File(Objects.requireNonNull(System.getProperty(EXTERNAL_SERVER_PROPERTY_NAME)));
-            final File[] externalServers = Objects.requireNonNull(externalDirectory.listFiles(file -> file.getName().endsWith(".jar")));
-            Assertions.assertEquals(1, externalServers.length);
-            jar = externalServers[0];
+        if (serverJar == null) {
+            final List<File> externalServers = getAvailableServers();
+            Assertions.assertEquals(1, externalServers.size());
+            jar = externalServers.get(0);
         } else {
-            jar = new File(jarName);
+            jar = serverJar;
         }
         Assertions.assertTrue(jar.exists(), "Jar could not be found " + jar.getAbsolutePath());
```

Comment:
Related improvement:
Perhaps we should move 
```
this.version = getVersion(jar)
```
Above starting the process.

Reply:
Done.